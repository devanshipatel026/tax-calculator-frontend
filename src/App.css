.App {
  text-align: left;
  background-color: white;
  width: 100%;
  position: absolute;
  height: 100%;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: rgb(87, 92, 92);
  min-height: 10vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(20px + 2vmin);
  color: white;
  margin-bottom: 0px;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.Home-text {
  align-content: left;
  font-weight: 100;
  font-family: Arial, Helvetica, sans-serif;
}

.Home-button {
  align-content: right;
}

.table {
  font-weight: bold;
}

table,
tr {
  text-align: left;
}
