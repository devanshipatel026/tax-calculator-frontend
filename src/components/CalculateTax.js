import React, { Component } from "react";
import { withRouter } from "react-router-dom";

class CalculateTax extends Component {
  constructor(props) {
    super(props);
    this.state = {
      income: "",
    };
  }

  goBack(e) {
    e.preventDefault();
    this.props.history.push({
      pathname: "/",
      state: { income: this.props.location.state.income },
    });
  }

  render() {
    return (
         
      <div>
        <button
          type="button"
          className="btn btn-outline-secondary"
          onClick={this.goBack.bind(this)}
        >
          Back
        </button>
        <table className="table mt-5 text-center">
          <thead className="thead-light">
            <tr>
              <th>Federal Tax Bracket</th>
              <th>Federal Tax Rates</th>
              <th>Federal Tax Amount Applicable</th>
            </tr>
          </thead>
          <tbody>
          {this.props.location.state.bracket.map(( taxBracket, index ) => (
            <tr key={index}>
              <td>{taxBracket.title}</td>
              <td>{taxBracket.percentage}%</td>
              <td>${taxBracket.taxAmount}</td>
            </tr>
          ))}
          <tr>
            <td colSpan="2" style={{"textAlign":"right"}}>Total Federal Tax:</td>
            <td>${this.props.location.state.totalTax}</td></tr>
          </tbody>
        </table>
      </div>
      );
  }
}

export default withRouter(CalculateTax);
