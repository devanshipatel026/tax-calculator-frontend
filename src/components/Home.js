import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import { getCalculatedTax } from "../services/tax";

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      income: "",
      fields: {},
      errors: {}
    };
  }

  submitForm(e) {
    e.preventDefault();
    getCalculatedTax(this.state.income).then((response) =>{ 
    if(response.message === 'Success'){
      this.props.history.push({
        pathname: "/CalculateTax",
        state: { income: this.state.income, bracket: response.bracket, totalTax: response.totalTax },
      });
    }else(
      console.log("Error retrieving Tax bracket!")
    )
  });
  }

  render() {
    return (
      <div>
        <form className="Home" onSubmit={this.submitForm.bind(this)}>
          <div className="form-group row">
            <label className="Home-text col-sm-2 col-form-label mt-2">
              Enter Gross Income:
            </label>
            <div className="col-sm-6 form-group row">
              <input
                required
                className="form-control mt-2"
                id="income"
                type="number"
                placeholder="Enter Gross Income (in CAD)"
                value={this.state.income}
                onChange={(e) => this.setState({ income: e.target.value })}
              />
              <button type="submit" className="btn btn-info mt-4">
                Calculate
              </button>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

export default withRouter(Home);
