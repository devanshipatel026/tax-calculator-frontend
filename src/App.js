import React, { Component } from "react";
import Home from "./components/Home";
import CalculateTax from "./components/CalculateTax";
import "./App.css";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";

class App extends Component {
  render() {
    return (
      <Router>
        <div>
          <h1 className="App-header">Tax Calculator</h1>
        </div>
        <div className="App">
          <Switch>
            <Route exact path="/" component={Home} />
            <Route path="/CalculateTax" component={CalculateTax} />
          </Switch>
        </div>
      </Router>
    );
  }
}
export default App;
