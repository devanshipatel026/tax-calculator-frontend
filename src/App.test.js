import React from 'react';
import { shallow, render, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import App from './App';
import Home from './components/Home';
import CalculateTax from './components/CalculateTax';
import renderer from 'react-test-renderer';

it("renders App without crashing", () => {
   shallow(<App />);
 });

 it("renders Home without crashing", () => {
  shallow(<Home />);
});

it("renders CalculateTax without crashing", () => {
  shallow(<CalculateTax />);
});

describe('App', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});