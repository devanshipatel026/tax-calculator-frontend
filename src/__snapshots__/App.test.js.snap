// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
Array [
  <div>
    <h1
      className="App-header"
    >
      Tax Calculator
    </h1>
  </div>,
  <div
    className="App"
  >
    <div>
      <form
        className="Home"
        onSubmit={[Function]}
      >
        <div
          className="form-group row"
        >
          <label
            className="Home-text col-sm-2 col-form-label mt-2"
          >
            Enter Gross Income:
          </label>
          <div
            className="col-sm-6 form-group row"
          >
            <input
              className="form-control mt-2"
              id="income"
              onChange={[Function]}
              placeholder="Enter Gross Income (in CAD)"
              required={true}
              type="number"
              value=""
            />
            <button
              className="btn btn-info mt-4"
              type="submit"
            >
              Calculate
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
]
`;
